shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -0,0 +1,323 @@
+use crate::frame::{CassandraOperation, CassandraResult, Frame};
+use crate::message::{IntSize, Message, MessageValue};
+use crate::{
+    config::topology::TopicHolder,
+    error::ChainResponse,
+    transforms::{Transform, Transforms, Wrapper},
+};
+use anyhow::Result;
+use async_trait::async_trait;
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, Clone, Copy)]
+pub enum RewriteConfig {
+    Port(u32),
+    EmulateSingleNode,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct CassandraPeersRewriteConfig {
+    pub rewrite: RewriteConfig,
+}
+
+impl CassandraPeersRewriteConfig {
+    pub async fn get_source(&self, _topics: &TopicHolder) -> Result<Transforms> {
+        Ok(Transforms::CassandraPeersRewrite(CassandraPeersRewrite {
+            rewrite: self.rewrite,
+        }))
+    }
+}
+
+#[derive(Clone)]
+pub struct CassandraPeersRewrite {
+    rewrite: RewriteConfig,
+}
+
+#[async_trait]
+impl Transform for CassandraPeersRewrite {
+    async fn transform<'a>(&'a mut self, message_wrapper: Wrapper<'a>) -> ChainResponse {
+        // Find the indices of queries to system.peers & system.peers_v2
+        let system_peers = message_wrapper
+            .messages
+            .iter()
+            .enumerate()
+            .filter(|(_, m)| is_system_peers(*m))
+            .map(|(i, _)| i)
+            .collect::<Vec<_>>();
+
+        let mut response = message_wrapper.call_next_transform().await?;
+
+        for i in system_peers {
+            match self.rewrite {
+                RewriteConfig::Port(new_port) => {
+                    rewrite_port(&mut response[i], new_port);
+                }
+                RewriteConfig::EmulateSingleNode => {
+                    emulate_single_node(&mut response[i]);
+                }
+            }
+        }
+
+        Ok(response)
+    }
+}
+
+fn is_system_peers(message: &Message) -> bool {
+    if let Frame::Cassandra(_) = message.original {
+        if let Some(namespace) = message.namespace() {
+            if namespace.len() > 1 {
+                return namespace[0] == "system"
+                    && (namespace[1] == "peers" || namespace[1] == "peers_v2");
+            }
+        }
+    }
+
+    false
+}
+
+/// Emulate a single node by removing the rows from a query to system.peers(_v2)
+/// Only Cassandra queries to the `system.peers` table found via the `is_system_peers(_v2)` functions should be passed to this
+fn emulate_single_node(message: &mut Message) {
+    if let Frame::Cassandra(ref mut frame) = message.original {
```

Comment:
`ref mut frame` syntax was superceded by `&mut message.original` syntax.
So it is preferred to write as:
`if let Frame::Cassandra(frame) = &mut message.original {`